### PR TITLE
Rewrote Default Template

### PR DESCRIPTION
**Pull request by vin-ni**
Rewrote the whole Template Structure.

The `DefaultTemplate.vue` file is the root of the template structure. Inside there is the `GraphPlayer.vue`. This allows for simple styling updates if this is just duplicated. If one needs a very different kind of player, the `GraphPlayer.vue` can be exchanged as well.

- "ready" trigger once the landing card is visible
- "start" trigger once clicked on start
- optional infos

### CODE REVIEW

File: caster-front/src/components/GraphPlayer.vue [hunk 1]
```diff
@@ -29,17 +29,42 @@ import { useSendStreamVariableMutation } from "@/graphql";
...
+    Graph,
+    | "uuid"
+    | "name"
+    | "aboutText"
+    | "displayName"
+    | "startText"
+    | "endText"
+    | "slugName"
+  >;
+  showDebug?: boolean;
```
Comment (capital-G) [inline caster-front/src/components/GraphPlayer.vue:56]: making `showDebug` optional via the `?` seems counter-intuitive for a boolean flag as it turns `true/false` to `true/false/undefined`.

Maybe defaulting it to `false` is a better way, see https://vuejs.org/guide/typescript/composition-api.html#typing-component-props

But I think it is not possible to apply partial defaults (e.g. we don't want to provide a default for `graph`).

In general only a remark for future development :)

File: caster-front/src/components/GraphPlayer.vue [hunk 2]
```diff
@@ -267,14 +292,40 @@ const convertButtonType = (b: ButtonType): ElButtonType => {
...
+const sendStreamVariableMutation = useSendStreamVariableMutation();
+
+// template functions
+const startStream = async () => {
+  // TODO: Do we still need this?
+  // I don't think so, since the page should show an error if it can't get a stream
+  // if (data.value?.streamInfo.__typename === "NoStreamAvailable") {
+  //   ElMessage.error("Can not start an unassigned stream");
+  //   return;
+  // }
```
Comment (capital-G) [inline caster-front/src/components/GraphPlayer.vue:305]: I sometimes added these kind of things in order to access the non `NoStreamAvailable` data of our subscription - I think it is safe to remove it :)

**Review by capital-G — APPROVED** [review]
Looks very nice :) Only some minor remarks that are optional or can be considered for future work :)

**Comment by capital-G** [comment]
#516 mentions that the drifter template is made obsolete by this PR?
In this case we should remove it within this PR as it creates less code to manage/test :)

**Comment by vin-ni** [comment]
Yes please.
Could you take it out of the backend? I'll remove it from the frontend.

**Comment by capital-G** [comment]
Is removed via https://github.com/Gencaster/gencaster/pull/516/commits/92856b12a73d039931eb47920614ef47ff7e6fbd now but needs a restart of the docker containers because the database needs a migration :) Haven't checked it myself now b/c it would conflict with the logging branch that I am working on rn

**Comment by vin-ni** [comment]
Checks running through. Ready to merge from my side

**Pull request merged**